client/me/purchases/list/site/index.jsx

```diff
@@ -14,6 +14,7 @@ import SectionHeader from 'components/section-header';
 
 const PurchasesSite = ( { isPlaceholder, name, purchases, slug } ) => {
 	let items, label = name;
+	const siteUrl = slug && slug.replace( /::/g, '/' );
```

Comment:
The problem was that site was being mapped to a new object in `getPurchasesBySite`.